src/cose/decoder.rs

```diff
@@ -113,25 +102,21 @@ pub fn decode_signature(bytes: Vec<u8>, payload: &[u8]) -> Result<CoseSignatures
 
     // Parse the protected signature header.
     let protected_signature_header = decode(protected_signature_header_bytes.clone())?;
-    // This isn't quite right due to order... (it's a map)
-    let signature_algorithm = match protected_signature_header {
+    // TODO: This isn't quite right due to order... (it's a map)
```

Comment:
Now that it is a map, the order isn't an issue. Maybe we should check for unknown key types, though?